cmd/pdatagen/internal/base_fields.go

```diff
@@ -462,3 +474,77 @@ func (pf *primitiveAsDoubleField) generateCopyToValue(sb *strings.Builder) {
 }
 
 var _ baseField = (*primitiveField)(nil)
+
+type field struct {
+	name            string
+	defaultVal      string
+	testVal         string
+	returnType      string
+	originFieldName string
+	originFullName  string
+	asName          string
+}
+type numberField struct {
```

Comment:
Rename: `oneofField`

Reply:
This would conflict w/ the existing `oneofField`, are you suggesting this would replace that?

Reply:
Eventually yes, but let's not resolve the world now :)), so maybe keep the current name number... and then we can look into replacing that :)